static/js/search.js

```diff
@@ -46,9 +49,9 @@ function initLunr() {
  * @param  {String} query
  * @return {Array}  results
  */
-function search(query) {
+function search(queryTerm) {
     // Find the item in our index corresponding to the lunr one to have more info
-    return lunrIndex.search(query).map(function(result) {
+    return lunrIndex.search(queryTerm+"^100"+" "+queryTerm+"*^10"+" "+"*"+queryTerm+"^10"+" "+queryTerm+"~2^1").map(function(result) {
```

Comment:
Could you explain this one for me?

Reply:
It makes the type-ahead style search work while making sure that closer the match is higher the score it gets.
https://github.com/olivernn/lunr.js/issues/256#issuecomment-295407852

Oliver's comments:
  **// look for an exact match and apply a large positive boost**
  q.term(queryTerm, { usePipeline: true, boost: 100 })

  **// look for terms that match the beginning of this queryTerm and apply a medium boost**
  q.term(queryTerm + "*", { usePipeline: false, boost: 10 })

  **// look for terms that match with an edit distance of 2 and apply a small boost**
  q.term(queryTerm, { usePipeline: false, editDistance: 2, boost: 1 })

I also added another option to match the terms which have the queryTerm at the end: "*"+queryTerm+"^10"
This would match the term background-color if you search for color.
It also improves matches in general. This new version doesn't archive every word, but pieces. I wasn't able to match some words because they weren't archived the way I searched them. I think it's better to have a false match down in the list than having no match at all.
One thing that would be nice is a **Search Results** page.